### ISSUE TITLE
File editor wont load files without an internet connection

### ISSUE THREAD

**Issue by magiva** [home-assistant/addons#3699]
### Describe the issue you are experiencing

I load file editor, select a file to load and nothing loads
the editor looks weird too with a blank horizontal bar top right, i can type in it but it doesnt do anything

![image](https://github.com/user-attachments/assets/376c34fe-845c-4162-b0a7-e80662f800a8)

Log:
INFO:2024-07-24 14:16:37,138:hass_configurator.configurator:172.30.32.2 - "GET /api/listdir?path=. HTTP/1.1" 200 -
INFO:2024-07-24 14:16:38,200:hass_configurator.configurator:127.0.0.1 - "GET / HTTP/1.1" 200 -
INFO:2024-07-24 14:16:40,234:hass_configurator.configurator:172.30.32.2 - "GET /api/file?filename=/homeassistant/configuration.yaml HTTP/1.1" 200 -

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

File editor

### What is the version of the add-on?

5.8.0

### Steps to reproduce the issue

1. open file editor
2. select file to open
3. observe that nothing has been loaded
...

### System Health information

There are currently no repairs available

### Anything in the Supervisor logs that might be useful for us?

```txt
nothing and nothing related to file, editor or add-on
```

**Comment by magiva** [comment]
i found the reason to be that i block cloudflare.com

![image](https://github.com/user-attachments/assets/03356112-8408-4e84-b41c-85efde84eef9)

as soon as i set it as trusted. it loads.
why would i need to have outside access to cloudflare to be able load locally ?

**Comment by Likqez** [comment]
Hi, I didn't find the source code for the file-editor add-on here on GitHub. But if it indeed is OSS, you could potentially remove the dependencies by serving the Cloudflare JS locally. As you stated, The functionality shouldn't require an active connection to 3rd parties.

It's also possible that the cdn portions are from https://github.com/ajaxorg/ace, which is used to power the File-editor.

Good Luck.

**Comment by Likqez** [comment]
> Edit: Add-on info shows https://github.com/home-assistant/addons/tree/master/configurator as source.

Yes, but do you actually see any source in there?
I did some more digging and it seems like [this](https://github.com/danielperna84/hass-configurator) is the UI portion of the source.

**Comment by Likqez** [comment]
Yes @OdinVex, but taking a look at the offical repositories `Dockerfile` you see that they uses `pip3 install hass-configurator` which is the repo I mentioned. Our problem is from the UI, the official repository however doesn't have any source code representing the UI.

TBH. it's just a shitty documented repository, where you actually have to dig to find out what dependencies they are using.

**Comment by Likqez** [comment]
### TLDR:

This is a **duplicate** of: https://github.com/danielperna84/hass-configurator/issues/177 which describes the situation! Check that out.

The **Source code** responsible for the UI and the external CDN requests:  [danielperna84/hass-configurator](https://github.com/danielperna84/hass-configurator)

**WIP build**(but stale) trying to get rid of CDN dependencies on the [offline](https://github.com/danielperna84/hass-configurator/tree/offline) Branch.

**Comment by starsoccer** [comment]
+1 would be great to get this working totally offline

**Comment by frenck** [comment]
As per above, this is an issue with/feature request for the upstream hass-configurator. Please track the issue on their end instead.

Thanks! 👍 

../Frenck

**Comment by magiva** [comment]
hilarious

**Comment by frenck** [comment]
@magiva Sorry?
